pyvista/plotting/renderer.py

```diff
@@ -732,20 +732,27 @@ def add_actor(
             Actor properties.
         """
         # Remove actor by that name if present
+        rv = None
         if name and remove_existing_actor:
             rv = self.remove_actor(name, reset_camera=False, render=False)
 
-        if isinstance(uinput, _vtk.vtkMapper):
-            actor = Actor()
-            actor.SetMapper(uinput)
-        else:
-            actor = uinput
+        if isinstance(actor, _vtk.vtkMapper):
+            actor = Actor(mapper=actor, name=name)
 
-        self.AddActor(actor)
-        actor.renderer = proxy(self)
+        if isinstance(actor, Actor) and name:
+            # WARNING: this will override the name if already set on Actor
+            actor.name = name
+
+        if isinstance(actor, Actor):
+            actor.renderer = proxy(self)
```

Comment:
I'm wondering why the `renderer` might want to be tracked on the `Actor`? Actors are often in multiple renderers. 

cc @akaszynski?

Reply:
It doesn't. Removed in [89475c0](https://github.com/pyvista/pyvista/pull/3864/commits/89475c045d0c4a3fdd3182f9343693e9030522e8).